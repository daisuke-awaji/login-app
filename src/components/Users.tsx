import React from 'react'
import { Link } from 'react-router-dom'

export function Users() {
  const users = [
    { id: 1, name: 'John', email: 'john@email.com' },
    { id: 2, name: 'Bob', email: 'bob@email.com' },
    { id: 3, name: 'Mark', email: 'mark@email.com' },
  ]
  const listUsers = users.map((user) => (
    <li key={user.id}>
      <Link to={'/users/' + user.id}>{user.name}</Link>
    </li>
  ))

  return (
    <div>
      <h1>Users</h1>
      <ul>{listUsers}</ul>
    </div>
  )
}
