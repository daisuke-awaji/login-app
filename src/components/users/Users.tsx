import React, { useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import { fetchUsers } from 'apis/users'

import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableContainer from '@material-ui/core/TableContainer'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import { IUser } from './IUser'

export const Users = () => {
  const [users, setUsers] = useState<IUser[]>([])

  const fetchData = async () => {
    const users = await fetchUsers()
    setUsers(users)
  }
  useEffect(() => {
    fetchData()
  }, [])
  return (
    <>
      <Typography variant="h3">Users</Typography>
      <TableContainer component={Paper}>
        <Table style={{ tableLayout: 'auto' }} aria-label="user table">
          <TableHead>
            <TableRow>
              <TableCell style={{ width: '10%' }} align="right">
                Id
              </TableCell>
              <TableCell style={{ width: '30%' }} align="right">
                Name
              </TableCell>
              <TableCell style={{ width: '40%' }} align="right">
                Email
              </TableCell>
              <TableCell style={{ width: '10%' }} align="right">
                Type
              </TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {users.map((user) => (
              <TableRow key={user.name}>
                <TableCell style={{ width: '10%' }} align="right">
                  {user.id}
                </TableCell>
                <TableCell style={{ width: '30%' }} align="right">
                  <Link to={'/users/' + user.id}>{user.name}</Link>
                </TableCell>
                <TableCell
                  style={{
                    width: '40%',
                    maxWidth: '100px',
                    // textOverflow: 'ellipsis', // はみ出たテキストを切り取って ... を表示する
                    whiteSpace: 'nowrap',
                    overflow: 'scroll',
                  }}
                  align="right"
                >
                  {user.email}
                </TableCell>
                <TableCell style={{ width: '10%' }} align="right">
                  {user.type}
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </>
  )
}
