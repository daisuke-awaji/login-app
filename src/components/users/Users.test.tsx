import React from 'react'
import { Router } from 'react-router-dom'
import Users from './Users'
import renderer from 'react-test-renderer'

const setup = () => {
  // Routerコンポーネントに依存する hisotry は mock 化して注入する
  const mockHistory: any = { push: jest.fn(), location: {}, listen: jest.fn() }
  const component = (
    <Router history={mockHistory}>
      <Users />
    </Router>
  )
  return { component }
}

describe('スナップショット', () => {
  it('レンダリングができること', () => {
    const { component } = setup()
    const header = renderer.create(component)
    let tree = header.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
