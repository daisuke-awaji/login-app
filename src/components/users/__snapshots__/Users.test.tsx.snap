// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショット レンダリングができること 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h2"
  >
    Users
  </h2>,
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="user table"
      className="MuiTable-root makeStyles-table-1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-69 MuiTableCell-alignRight"
            scope="col"
            style={
              Object {
                "width": "10%",
              }
            }
          >
            Id
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-69 MuiTableCell-alignRight"
            scope="col"
            style={
              Object {
                "width": "30%",
              }
            }
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-69 MuiTableCell-alignRight"
            scope="col"
            style={
              Object {
                "width": "40%",
              }
            }
          >
            Email
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-69 MuiTableCell-alignRight"
            scope="col"
            style={
              Object {
                "width": "10%",
              }
            }
          >
            Type
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
          style={
            Object {
              "height": 265,
            }
          }
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            colSpan={6}
          />
        </tr>
      </tbody>
      <tfoot
        className="MuiTableFooter-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-footer"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root"
            colSpan={3}
          >
            <div
              className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
            >
              <div
                className="MuiTablePagination-spacer"
              />
              <p
                className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
              >
                Rows per page:
              </p>
              <div
                className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
                onClick={[Function]}
              >
                <select
                  aria-label="rows per page"
                  className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiInputBase-input"
                  multiple={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  value={5}
                >
                  <option
                    className="MuiTablePagination-menuItem"
                    value={5}
                  >
                    5
                  </option>
                  <option
                    className="MuiTablePagination-menuItem"
                    value={10}
                  >
                    10
                  </option>
                  <option
                    className="MuiTablePagination-menuItem"
                    value={25}
                  >
                    25
                  </option>
                  <option
                    className="MuiTablePagination-menuItem"
                    value={-1}
                  >
                    All
                  </option>
                </select>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
              >
                0-0 of 0
              </p>
              <div
                className="makeStyles-root-136"
              >
                <button
                  aria-label="first page |<"
                  className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="previous page <"
                  className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="next page >"
                  className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="last page >|"
                  className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>,
]
`;
